homeassistant/components/playstation_network/coordinator.py

```diff
@@ -0,0 +1,133 @@
+            self.data["presence"] = await self.hass.async_add_executor_job(
+                self.user.get_presence()
+            )
+            self.data["available"] = (
+                self.data["presence"].get("basicPresence").get("availability")
+                == "availableToPlay"
+            )
+            self.data["platform"] = (
+                self.data["presence"].get("basicPresence").get("primaryPlatformInfo")
+            )
+            try:
+                self.data["title_metadata"] = (
+                    self.data["presence"]
+                    .get("basicPresence")
+                    .get("gameTitleInfoList")[0]
+                )
+            except Exception:  # noqa: BLE001
+                self.data["title_metadata"] = {}
+
+            # self.data["friends"] = await self.client.available_to_play()
+            self.data["trophy_summary"] = await self.hass.async_add_executor_job(
+                self.client.trophy_summary()
+            )
+
+            self.data["country"] = self.hass.config.country
+            self.data["language"] = self.hass.config.language
+
+            if (
+                self.data["available"] is True
+                and self.data["title_metadata"].get("npTitleId") is not None
+            ):
+                title_id = self.data["title_metadata"].get("npTitleId")
+                title = await self.hass.async_add_executor_job(
+                    lambda: self.api.game_title(title_id, "me")
+                )
+                ## Attempt to pull details with user's country and language code
+                self.data["title_details"] = await self.hass.async_add_executor_job(
+                    lambda: title.get_details(
+                        self.hass.config.country, self.hass.config.language
+                    )
+                )
+                ## If we receive an error, fall back to english
+                if self.data["title_details"][0].get("errorCode") is not None:
+                    self.data["title_details"] = await self.hass.async_add_executor_job(
+                        title.get_details()
+                    )
+
+                trophy_titles = await self.hass.async_add_executor_job(
+                    lambda: self.client.trophy_titles_for_title(
+                        title_ids=[title.title_id]
+                    )
+                )
+                await self.hass.async_add_executor_job(
+                    lambda: self._get_trophies(trophy_titles)
+                )
+
+                titles_with_stats = await self.hass.async_add_executor_job(
+                    lambda: self.client.title_stats(limit=3, page_size=3)
+                )
+                await self.hass.async_add_executor_job(
+                    lambda: self._get_titles(titles_with_stats)
+                )
+
+            return self.data  # noqa: TRY300
+        except PSNAWPAuthenticationError as error:
+            raise ConfigEntryAuthFailed(error) from error
```

Comment:
```suggestion
            raise UpdateFailed(error) from error
```
Without a reauth flow this will fail.